### PR TITLE
Make sure .setup() runs on dynamic services.

### PR DESCRIPTION
**Pull request by marshallswain**
Setup() wasn't being run on dynamic services.  I've included a fix and updated one of the tests to make sure setup runs.

Oh, I also moved `stripSlashes(location)` to the top of app.service() and replaced all duplicate instances of it with `location`.

### CODE REVIEW

**Comment by marshallswain** [comment]
@daffl Does this look ready to merge?

File: test/providers/primus.test.js [hunk 1]
```diff
@@ -438,10 +438,16 @@ describe('Primus provider', function () {
       socket.send('tasks::remove', 1, {}, function() {});
       socket.send('tasks::remove', 23, {}, function() {});
 
+      var ready = false;
+
       socket.on('tasks removed', function (data) {
         service.removed = oldRemoved;
         assert.equal(data.id, 23);
-        done();
+        console.log('Ready: ' + ready);
```
Comment (daffl) [inline test/providers/primus.test.js]: `console.log`

**Comment by daffl** [comment]
Yes, after you remove the `console.log` and please squash the commits into one (http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html - but I can to that, too before merging).

**Comment by marshallswain** [comment]
Wow.  squash is new for me.  That was a beautiful experience.

**Pull request merged**